File: packages/react-native-nitro-image/ios/HybridViewWithChildren.swift
//
//  HybridViewWithChildren.swift
//  NitroImage
//
//  Created by Patrick Kabwe on 06/05/2025.
//

import Foundation
import UIKit

class HybridViewWithChildren : HybridViewWithChildrenSpec {
  // UIView
  var view: UIView = UIView()

  // Props
  var isBlue: Bool = false {
    didSet {
      view.backgroundColor = isBlue ? .systemBlue : .systemRed
    }
  }
  var colorScheme: ColorScheme = .light
  var someCallback: () -> Void = { }

  // Methods
  func someMethod() throws -> Void {
    someCallback()
  }
}

File: packages/react-native-nitro-test/android/src/main/java/com/margelo/nitro/test/HybridViewWithChildren.kt
package com.margelo.nitro.test;

import android.content.Context
import android.graphics.Color
import androidx.annotation.Keep;
import com.facebook.proguard.annotations.DoNotStrip;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.views.view.ReactViewGroup


class ViewWithChildrenImpl(context: Context) : ReactViewGroup(context) {}

@Keep
@DoNotStrip
class HybridViewWithChildren(val context: ThemedReactContext): HybridViewWithChildrenSpec() {
    // View
    override val view = ViewWithChildrenImpl(context)

    // Props
    override var colorScheme: ColorScheme = ColorScheme.LIGHT
    override var someCallback: () -> Unit = {}

    // Methods
    override fun someMethod(): Unit {
        someCallback()
    }
}